week06/employee_test.go

```diff
@@ -0,0 +1,85 @@
+package week06
+
+import (
+	"testing"
+
+)
+
+func TestEmployeeIsValid(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name	string
+		id		int
+		err 	error
+	}{
+		{name: "test invalid employee", id: -1, err: ErrInvalidEmployee(-1)},
+		{name: "test invalid employee", id: 0, err: ErrInvalidEmployee(0)},
+		{name: "test valid employee", id: 1, err: nil},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			emp := Employee(tc.id)
+
+			act := emp.IsValid()
+			exp := tc.err
+
+			if exp != act {
+				t.Fatalf("expected %s, got %s", exp, act)
+			}
+		})
+	}
+}
+
+func TestEmployeeWork(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name	string
+		id		int
+		err 	error
+	}{
+		{name: "test invalid employee", id: 0, err: ErrInvalidEmployee(0)},
+		{name: "test valid employee", id: 1, err: nil},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			m := NewManager()
```

Comment:
Always make sure to clean up resources. The `m.Stop` method is only called on a happy path. Every time you mark the test as failed, you are leaking resources.

Defer the `m.Stop` method so it always gets run with the test exists.

```suggestion
			m := NewManager()
			defer m.Stop()
```